File: Cpp/Algorithm/src/Chapter3/twoSum_1.cpp
#include "chapter3.h"
using namespace myAlgo;
vector<int> TwoSum_1::twoSum(vector<int>& nums, int target)
{
	unordered_map<int, int> hashtable;
	for (int i = 0; i < nums.size(); ++i) {
		auto it = hashtable.find(target - nums[i]);
		if (it != hashtable.end()) {
			return { it->second, i };
		}
		hashtable[nums[i]] = i;
	}
	return {};
}

File: JavaAlgorithm/src/main/java/algo/tzashinorpu/FourthRound/Chapter03/twoSum_1.java
package algo.tzashinorpu.FourthRound.Chapter03;

import java.util.HashMap;

public class twoSum_1 {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        HashMap<Integer, Integer> m = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (m.containsKey(target - nums[i])) {
                res[0] = m.get(target - nums[i]);
                res[1] = i;
            } else {
                m.put(nums[i], i);
            }
        }
        return res;
    }
}
